### PR TITLE
kie-issues#737: Test Scenario Editor: Background Tab

### LINKED ISSUE #737: Test Scenario Editor: Background Tab
The scope of this Task is to reimplement the Background Table of Test Scenario Editor.

Screenshot 2023-12-04 at 17 38 26

As this is a reduced version of the ScenarioGrid, the idea is to reuse the TestScenarioTable introduced here https://github.com/apache/incubator-kie-issues/issues/629, abstracting it to support both cases.

Blocked by:
- https://github.com/apache/incubator-kie-issues/issues/629

### PR DESCRIPTION
**Pull request by yesamer**
Closes https://github.com/apache/incubator-kie-issues/issues/737

This PR introduces the Background Table. 
The previously introduced TestScenario table was extended using a "generics" approach, to manage both cases.

Live Demo: https://yesamer.github.io/prototypes/scesim-editor/

This is the last PR managed as a multiple subtasks: Starting from now, I'll manage the whole scesim refactoring in a single PR.

### CODE REVIEW

**Comment by yesamer** [comment]
@tiagobento Ready to be reviewed / merged :)

**Review by tiagobento — APPROVED** [review]

File: packages/scesim-editor/src/table/TestScenarioTable.tsx [hunk 1]
```diff
@@ -81,11 +88,48 @@ function TestScenarioTable({
...
+  const retrieveRowsData = useCallback(
+    (rowData: SceSim__backgroundDatasType | SceSim__scenariosType) => {
+      if (isBackground) {
+        return (rowData as SceSim__backgroundDatasType).BackgroundData;
+      } else {
+        return (rowData as SceSim__scenariosType).Scenario;
+      }
+    },
+    [isBackground]
+  );
```
Comment (jomarko) [inline packages/scesim-editor/src/table/TestScenarioTable.tsx:110]: @yesamer thank you for the PR. Just out of interest, could you please explain the need of both `SceSim__backgroundDatasType` and `SceSim__scenariosType`?

They seems to be the same in xsd:
```
<xsd:complexType name="ScenarioType">
    <xsd:sequence>
      <xsd:element type="factMappingValuesType" name="factMappingValues" />
    </xsd:sequence>
  </xsd:complexType>
  <xsd:complexType name="BackgroundDataType">
    <xsd:sequence>
      <xsd:element type="factMappingValuesType" name="factMappingValues" />
    </xsd:sequence>
  </xsd:complexType>
  <xsd:complexType name="scenariosType">
    <xsd:sequence>
      <xsd:element type="ScenarioType" name="Scenario" maxOccurs="unbounded" minOccurs="0" />
    </xsd:sequence>
  </xsd:complexType>
  <xsd:complexType name="backgroundDatasType">
    <xsd:sequence>
      <xsd:element type="BackgroundDataType" name="BackgroundData" maxOccurs="unbounded"
        minOccurs="0" />
    </xsd:sequence>
```
    
    
And most of the code (pseudocode) seems to use `isbackground` that is implemented without the need of `SceSim__backgroundDatasType` or `SceSim__scenariosType`.
    
I mean we do most the time:
```
if (isBackground) {
  // do something
}
```

We do not do:
```
if (is SceSim__backgroundDatasType) {do sth}
else if (is SceSim__scenariosType) {do sth else}
```


Just trying to understand code better. I am not saying the current version is wrong.

Comment (yesamer) [reply packages/scesim-editor/src/table/TestScenarioTable.tsx:110]: Hi @jomarko. apologies for my late reply. You highlighted a very interesting point, I'll investigate more and try to check if your comment is applicable. Thx!

**Comment by yesamer** [comment]
@jomarko I applied the following change:

Screenshot 2024-01-16 at 10 51 03

Here, I merged `ScenarioType` and `BackgroundDataType` types on a single new type FactMappingValuesTypes, considering they hold the data as you correctly spotted.
Can't do the same for`scenariosType` & `backgroundDatasType` types: they hold the same data, but they are bound to different tag names, and that means I can't remove those `if` management. Does make sense? WDYT?

**Pull request merged**